### PR TITLE
Prep script for running the docs locally

### PR DESCRIPTION
**Pull request by nellicus**
This PR introduces 2 extra yarn commands to avoid the tedious task of:

```
git clone https://github.com/ClickHouse/ClickHouse
git clone https://github.com/ClickHouse/ClickHouse-docs

# Docusaurus expects all of the markdown files to be located in the directory tree clickhouse-docs/docs/.
# This is not the way our repos are set up, so some copying of files is needed to build the docs:
# Note: Symlinks will not work.
cp -r ClickHouse/docs/en/development     ClickHouse-docs/docs/en/
cp -r ClickHouse/docs/en/engines         ClickHouse-docs/docs/en/
cp -r ClickHouse/docs/en/getting-started ClickHouse-docs/docs/en/
cp -r ClickHouse/docs/en/interfaces      ClickHouse-docs/docs/en/
cp -r ClickHouse/docs/en/operations      ClickHouse-docs/docs/en/
cp -r ClickHouse/docs/en/sql-reference   ClickHouse-docs/docs/en/
cp -r ClickHouse/docs/ru                 ClickHouse-docs/docs/
cp -r ClickHouse/docs/zh                 ClickHouse-docs/docs/

cd ClickHouse-docs
```

when building the docs locally.

this 1 liner script expects the user to sit in local `Clickhouse/clickhouse-docs` root folder;

then running `yarn prep-from-master`:

- clones Clickhouse/Clickhouse repo to a randomly name `/tmp/ch_temp_${random_integer}` folder
- copies one by one the required folders into related docs repo folders;
- removes the temporary `/tmp/ch_temp_${random_integer}`

This PR also sorts the yarn scripts alphabetically and updates the contrib guide

Sample execution:
```
➜  clickhouse-docs git:(automate-doc-prep) ✗ yarn prep-from-master
yarn run v1.22.19
$ array_root=(docs/ru docs/zh);array_en=(docs/en/development docs/en/engines docs/en/getting-started docs/en/interfaces docs/en/operations docs/en/sql-reference);ch_temp=/tmp/ch_temp_$RANDOM && mkdir -p $ch_temp && git clone --depth 1 --branch master https://github.com/ClickHouse/ClickHouse $ch_temp; for folder in ${array_en[@]}; do cp -r $ch_temp/$folder docs/en;echo "Copied $folder from ClickHouse source";done;for folder in ${array_root[@]}; do cp -r $ch_temp/$folder docs/;echo "Copied $folder from ClickHouse source";done;rm -rf $ch_temp && echo "Prep completed";
Cloning into '/tmp/ch_temp_24972'...
remote: Enumerating objects: 22888, done.
remote: Counting objects: 100% (22888/22888), done.
remote: Compressing objects: 100% (18700/18700), done.
remote: Total 22888 (delta 3102), reused 10510 (delta 2106), pack-reused 0
Receiving objects: 100% (22888/22888), 75.40 MiB | 22.66 MiB/s, done.
Resolving deltas: 100% (3102/3102), done.
Updating files: 100% (23435/23435), done.
Copied docs/en/development from ClickHouse source
Copied docs/en/engines from ClickHouse source
Copied docs/en/getting-started from ClickHouse source
Copied docs/en/interfaces from ClickHouse source
Copied docs/en/operations from ClickHouse source
Copied docs/en/sql-reference from ClickHouse source
Copied docs/ru from ClickHouse source
Copied docs/zh from ClickHouse source
Prep completed
✨  Done in 38.97s.
➜  clickhouse-docs git:(automate-doc-prep) ✗ yarn install
yarn install v1.22.19
[1/5] 🔍  Validating package.json...
[2/5] 🔍  Resolving packages...
success Already up-to-date.
✨  Done in 0.18s.
➜  clickhouse-docs git:(automate-doc-prep) ✗ yarn start  
yarn run v1.22.19
$ docusaurus start

   ╭───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╮
   │                                                                                                                                                                                                   │
   │                                                                                  Update available 2.3.0 → 2.3.1                                                                                   │
   │                                                                                                                                                                                                   │
   │                                                         To upgrade Docusaurus packages with the latest version, run the following command:                                                        │
   │     `yarn upgrade @docusaurus/core@latest @docusaurus/plugin-client-redirects@latest @docusaurus/preset-classic@latest @docusaurus/theme-mermaid@latest @docusaurus/module-type-aliases@latest`   │
   │                                                                                                                                                                                                   │
   ╰───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╯

[INFO] Starting the development server...
[SUCCESS] Docusaurus website is running at: http://localhost:3000/docs/

✔ Client
  Compiled successfully in 49.32s

client (webpack 5.75.0) compiled successfully
```

also `prep-from-local /full/path/to/local/clickhouse/clickhouse` is added to allow copying from a local folder containing the clickhouse/clickhouse repo files.

```
➜  clickhouse-docs git:(automate-doc-prep) ✗ yarn prep-from-local /Users/abonuccelli/Projects/ClickHouse/ClickHouse 
yarn run v1.22.19
$ sh -c 'array_root=(docs/ru docs/zh);array_en=(docs/en/development docs/en/engines docs/en/getting-started docs/en/interfaces docs/en/operations docs/en/sql-reference);for folder in ${array_en[@]}; do cp -r $0/$folder docs/en;echo "Copied $folder from ClickHouse local path [$0]";done;for folder in ${array_root[@]}; do cp -r $0/$folder docs/;echo "Copied $folder from ClickHouse local path [$0]";done;echo "Prep completed";' /Users/abonuccelli/Projects/ClickHouse/ClickHouse
Copied docs/en/development from ClickHouse local path [/Users/abonuccelli/Projects/ClickHouse/ClickHouse]
Copied docs/en/engines from ClickHouse local path [/Users/abonuccelli/Projects/ClickHouse/ClickHouse]
Copied docs/en/getting-started from ClickHouse local path [/Users/abonuccelli/Projects/ClickHouse/ClickHouse]
Copied docs/en/interfaces from ClickHouse local path [/Users/abonuccelli/Projects/ClickHouse/ClickHouse]
Copied docs/en/operations from ClickHouse local path [/Users/abonuccelli/Projects/ClickHouse/ClickHouse]
Copied docs/en/sql-reference from ClickHouse local path [/Users/abonuccelli/Projects/ClickHouse/ClickHouse]
Copied docs/ru from ClickHouse local path [/Users/abonuccelli/Projects/ClickHouse/ClickHouse]
Copied docs/zh from ClickHouse local path [/Users/abonuccelli/Projects/ClickHouse/ClickHouse]
Prep completed
✨  Done in 0.40s.
```

@rfraposa @DanRoscigno

### CODE REVIEW

**Comment by nellicus** [comment]
updated the PR description with a sample execution.

I do get 

Screenshot 2023-03-03 at 17 27 44
Screenshot 2023-03-03 at 17 27 52

is the first one expected?

**Comment by DanRoscigno** [comment]
I think that this looks very useful.  Here is my typical workflow:

- rm -rf clickhouse-docs/docs
- git restore docs
- Edit files in both ClickHouse/ClickHouse and ClickHouse/clickhouse-docs
- cp -r all of the dirs from ClcikHouse/ClickHouse/docs/ into clickhouse-docs
- yarn start

Your yarn prep copies in the files from ClickHouse/ClickHouse master, but not from the files I edit in my branch.

Could you give an option to remove the existing files in the dirs from ClickHouse/ClickHouse, then copy files from a local fork of ClickHouse/ClickHouse?  A command something like:

yarn prep $DOCS/ClickHouse

**Comment by DanRoscigno** [comment]
> 
> is the first one expected?

Yes, we have redirects for `/docs`, but not for `/`  .  I think that the Docusaurus build process should include the baseURL in the URL that they launch the browser with, but they don't do that.

**Review by rfraposa — APPROVED** [review]
This is awesome - thanks Tony!

**Comment by nellicus** [comment]
@DanRoscigno as discussed I'm gonna add a separate prep command that allows copying the clickhouse/clickhouse/docs folder from a given local fs path (as opposed to using the locally cloned clickhouse/clickhouse/docs master branch) (what `yarn:prep` currently does)

**Comment by nellicus** [comment]
@DanRoscigno I've pushed the changes and updated the PR descriptions. the 2 commands:

```
yarn prep-from-master
yarn prep-from-local /full/path/to/local/clickhouse/clickhouse
```

should address the other use cases discussed.

Please test this thoroughly and let me know your feedback then if all is good I'll merge.

File: contrib-writing-guide.md [hunk 1]
```diff
@@ -33,28 +33,29 @@ The documentation is built with Docusaurus, which requires Node.js. We recommend
...
+yarn prep-from-master
+
+# OR
+
+# yarn prep-from-local
+# This command will use a locally available folder containing ClickHouse/Clickhouse
+# and from there it will copy over the relevant docs folders to this folder
+
+# running this without using the local full path as an argument will lead to error
+yarn prep-from-local /full/path/to/your/local/Clickhouse/Clickhouse
```
Comment (DanRoscigno) [inline contrib-writing-guide.md:55]:
```suggestion

yarn prep-from-local /full/path/to/your/local/Clickhouse/Clickhouse 
```

**Review by DanRoscigno — APPROVED** [review]
LGTM

Thanks @nellicus I tested with master and with a local branch.  Works great.

**Pull request merged**